From 2528693978ac5a240e1aae0a126607d80fbf114b Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@rr-research.no>
Date: Mon, 7 Oct 2019 12:22:52 +0200
Subject: [PATCH 11/30] COMP: Enable installation of generated .h files for
 BaseQTLibraries

Some SlicerBaseQT libraries generate ui_*.h files. With this, these
files will be also installed with the rest of development files
---
 CMake/SlicerMacroBuildBaseQtLibrary.cmake | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMake/SlicerMacroBuildBaseQtLibrary.cmake b/CMake/SlicerMacroBuildBaseQtLibrary.cmake
index 66cfc933cf..262756bb2f 100644
--- a/CMake/SlicerMacroBuildBaseQtLibrary.cmake
+++ b/CMake/SlicerMacroBuildBaseQtLibrary.cmake
@@ -141,6 +141,9 @@ macro(SlicerMacroBuildBaseQtLibrary)
     set(_moc_options OPTIONS -DSlicer_HAVE_QT5)
     QT5_WRAP_CPP(SLICERQTBASELIB_MOC_OUTPUT ${SLICERQTBASELIB_MOC_SRCS} ${_moc_options})
     QT5_WRAP_UI(SLICERQTBASELIB_UI_CXX ${SLICERQTBASELIB_UI_SRCS})
+
+    message(STATUS "---------------->>> ${SLICERQTBASELIB_UI_SRCS}")
+    message(STATUS "================>>> ${SLICERQTBASELIB_UI_CXX}")
     if(DEFINED SLICERQTBASELIB_RESOURCES)
       QT5_ADD_RESOURCES(SLICERQTBASELIB_QRC_SRCS ${SLICERQTBASELIB_RESOURCES})
     endif()
@@ -239,6 +242,11 @@ macro(SlicerMacroBuildBaseQtLibrary)
       ${dynamicHeaders}
       DESTINATION ${Slicer_INSTALL_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
       )
+
+    install(FILES
+      ${SLICERQTBASELIB_UI_CXX}
+      DESTINATION ${Slicer_INSTALL_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
+      )
   endif()
 
   # --------------------------------------------------------------------------
-- 
2.26.2

